fabfile.py

```diff
@@ -0,0 +1,68 @@
+import os
+import yaml
+
+from fabric.api import env, run
+
+
+# Load in server config, here's a quick example:
+#   role_name:
+#     hosts:
+#       - ubuntu@12.34.56.78
+#     broker_url: pyamqp://user:pass@host:port/vhost-gpu
+#     is_gpu: true
+#     docker_image: codalab/competitions-v2-compute-worker:nvidia
+#
+# You select the role to run with like s:
+#   $ fab -R role_name <command>
+env.roledefs = yaml.load(open('server_config.yaml').read())
+
+
+def status():
+    """Gets status of all docker containers on server"""
+    run("docker ps")
+
+
+def update():
+    """Updates docker image on server and restarts the worker, based on server_config.yaml settings.
+
+    See README (Worker management section) for example settings."""
+    # Ensure that we're referencing one and only 1 role, so we don't accidentally update CPUs to GPUs
+    if not env.effective_roles:
+        print("ERROR: You must specify a role when running this task. I.e. fab -R some-gpu task_name`")
+        return
+    if len(env.effective_roles) > 1:
+        print("ERROR: Only specify 1 role (because we need 1 broker_url) when running this task")
```

Comment:
can we clean up some of the comments and verbiage in this doc to make it more clear that a "role" is either a cpu or gpu command? Role on its own is a little confusing.

Reply:
It's not either cpu/gpu though it could be any role you define from `server_config.yaml` -- each of those is a "role"

hmm..

Reply:
can leave it as is I suppose. I still think it is unintuitive, but it sounds like fabric is in general at the moment?

Reply:
fabric v2 is unintuitive for sure, this is fabric v1 though which only took a few minutes to setup. I think roledefs isn't that wacky to understand, but maybe I'm just used to it. 

```yaml
role_name:
  hosts:
    - user@ip
    - user@ip2
```

```bash
# to run a task against all hosts within role_name
$ fab -R role_name task_name
```

Maybe I could define role better somewhere? will think about this for a sec